### PR TITLE
Add xrt::version APIs

### PR DESCRIPTION
**Pull request by stsoe**
#### Problem solved by the commit
Ability to check XRT version and build at run-time.

### CODE REVIEW

File: src/runtime_src/core/common/api/xrt_version.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+unsigned int
+build()
+{
+  return XRT_BUILD;
+}
+
+} // namespace xrt::version
+
+////////////////////////////////////////////////////////////////
+// xrt_message C API implmentations (xrt_version.h)
```
Comment (keryell) [inline src/runtime_src/core/common/api/xrt_version.cpp:46]:
```suggestion
// xrt_version C API implementations (xrt_version.h)
// with "C" linkage.
```

Comment (stsoe) [reply src/runtime_src/core/common/api/xrt_version.cpp]: Duh.

**Comment by stsoe** [comment]
retest this please - seems to build correctly when using build_edge -aarch versal locally

**Comment by xilinxgitops** [comment]
retest this please - debugging issue with Vamshi

**Comment by keryell** [comment]
I am discovering this `sys/sysmacros.h` nightmare. :-(
Is `sys/types.h` required by the XRT API? Perhaps this API enabling file-descriptor sharing?
Could this plague contained in only the internal `.c`/`.cpp` files using it?

**Comment by stsoe** [comment]
> Is `sys/types.h` required by the XRT API? Perhaps this API enabling file-descriptor sharing?
> Could this plague contained in only the internal `.c`/`.cpp` files using it?

I admit I didn't track down where this include came from, but xrt_version.cpp does not include (implicitly) anything but `<string>` and `<iostream>` .   

I suspect iostream is the culprit, and yes, I don't really fancy `iostream` being included especially not in a header file, but that's unfortunately how the version.h file generated by CMake was coded up.  Let me revisit that, and cleanup src/CMake/config/version.h.in and move the definition of the print function into the xrt_version.cpp.  While a proper change, this doesn't actually solve the major minor macro mess.

**Comment by manikandan-xilinx** [comment]
retest this please. (Including hidden files during checkout)

**Comment by keryell** [comment]
Curious.
I cannot imagine C++ `<iostream>` or anything else pulling anything from `sys/` defining some atrocious macros like that.

**Comment by stsoe** [comment]
Well I don't know, but as I said, the macro mess is isolated to `xrt_version.cpp` which includes `gen/version.h` which in turn includes `<string>` and `<iostream>`

**Pull request merged**